SingularityService/src/main/java/com/hubspot/singularity/auth/SingularityAuthorizationHelper.java

```diff
@@ -138,13 +143,17 @@ public void checkForAuthorization(SingularityRequest request, Optional<Singulari
     checkUnauthorized(user.isPresent(), "user must be present");
 
     final Set<String> userGroups = user.get().getGroups();
+    final Set<String> readWriteGroups = request.getReadWriteGroups().or(Collections.<String>emptySet());
+    if (request.getGroup().isPresent()) {
+      readWriteGroups.add(request.getGroup().get());
```

Comment:
No worries -- you can still use `Sets.union()` by coercing the `Optional<>` to a `Set<>` via `request.getGroup().asSet()`